lib/thrift/generator/client.ex

```diff
@@ -0,0 +1,125 @@
+defmodule Thrift.Generator.Client do
+  defmodule Framed do
+    alias Thrift.Generator.Service
+    alias Thrift.Parser.Models.Function
+    alias Thrift.Clients.BinaryFramed
+
+    def generate(service_module, service) do
+      functions = service.functions
+      |> Map.values
+      |> Enum.map(&generate_handler_function(service_module, &1))
+
+      quote do
+        defmodule Client.Framed do
+          alias Thrift.Clients.BinaryFramed
+          alias Thrift.Protocols.Binary
+
+          defdelegate close(conn), to: BinaryFramed
+          defdelegate connect(conn, opts), to: BinaryFramed
+
+          def start_link(host, port, tcp_opts, timeout \\ 5000) do
+            BinaryFramed.start_link(host, port, tcp_opts, timeout)
+          end
+          unquote_splicing(functions)
+        end
+      end
+    end
+
+    defp generate_handler_function(service_module, function) do
+
+      args_module = service_module
+      |> Module.concat(Service.service_module_name(function, :args))
+
+      response_module = Service.service_module_name(function, :response)
+
+      underscored_name = function.name
+      |> Atom.to_string
+      |> Macro.underscore
+      |> String.to_atom
+
+      underscored_options_name = :"#{underscored_name}_with_options"
+      bang_name = :"#{underscored_name}!"
+      options_bang_name = :"#{underscored_options_name}!"
+
+      vars = function.params
+      |> Enum.map(&Macro.var(&1.name, nil))
+
+      assignments = function.params
+      |> Enum.zip(vars)
+      |> Enum.map(fn {param, var} ->
+        quote do
+          {unquote(param.name), unquote(var)}
+        end
+      end)
+
+      rpc_name = Atom.to_string(function.name)
+
+      def_type = if function.oneway do
+        quote do: defp
+      else
+        quote do: def
+      end
+
+      quote do
+
+        unquote(def_type)(unquote(underscored_options_name)(client, unquote_splicing(vars), opts)) do
+          serialized_args = %unquote(args_module){unquote_splicing(assignments)}
+          |> unquote(args_module).BinaryProtocol.serialize
+
+          sequence_id = :erlang.unique_integer([:positive])
```

Comment:
Can you say why this need to be a generated value as opposed to a simple counter value stored in the client's state?